Documentation/ceph-common-issues.md

```diff
@@ -953,3 +953,79 @@ If you are using Kubernetes version is v1.15 or older, you will see an error lik
+
+
+## Unexpected partitions created
+
+### Symptoms
+**Rook versions v1.6.0-v1.6.6 may create unwanted OSDs on partitions that appear unexpectedly and
+seemingly randomly, which will corrupt existing OSDs.**
+
+Unexpected partitions are created on host disks that are used by Ceph OSDs. This happens more often
+on SSDs than HDDs and usually only on disks that are 1TB or larger. Many tools like `lsblk`,
+`blkid`, and `udevadm` will not show a partition table type for the partition; however, `parted`
+will reliably list the partition as `atari` (see the example command and output below).
+
+```console
+# parted --script /dev/sdb1 print
+Model: Unknown (unknown)
+Disk /dev/sdb1: 1509kB
+Sector size (logical/physical): 512B/512B
+Partition Table: atari
+Disk Flags:
+
+Number  Start  End  Size  Type  File system  Flags
+
+```
+
+The underlying issue causing this is Atari (sometimes identified as AHDI) partition support in the
+Linux kernel. Atari partitions have very relaxed specifications compared to modern partition types,
+and it is relatively easy for random data written to a disk to appear as an Atari partition to the
+Linux kernel. Ceph's Bluestore OSDs have an anecdotally high probability of writing data on to disks
+that can appear to the kernel as an Atari partition.
+
+You can see https://github.com/rook/rook/issues/7940 for more detailed information and discussion.
+
+### Solution
+#### Recover from corruption (v1.6.0-v1.6.6)
+If you are using Rook v1.6, you must update to v1.6.7 or higher to avoid further incidents of OSD
+corruption caused by these Atari partitions.
+
+If there are already OSDs created on the unexpected partitions, OSD corruption has already occurred.
+To recover from this, immediately update to v1.6.7 or higher. After the update, no further OSD
+corruption should occur. Next, to get back to a healthy Ceph cluster state, focus on one corruped
+disk at a time and [remove all OSDs on each corrupted disk](ceph-osd-mgmt.md#remove-an-osd) one disk
+at a time.
+
+As an example, you may have `/dev/sdb` with two unexpected partitions (`/dev/sdb1` and `/dev/sdb2`)
+as well as a second corrupted disk `/dev/sde` with one unexpected partition (`/dev/sde1`).
+1. First, remove the 3 OSDs associated with `/dev/sdb`, `/dev/sdb1`, and `/dev/sdb2`.
+2. Then wipe clean `/dev/sdb`.
+3. After this, allow the Rook operator to deploy a new OSD to `/dev/sdb`; this will allow Ceph to
+   use `/dev/sdb` for data recovery and replication while the next OSDs are removed.
+4. Now Repeat steps 1-3 for `/dev/sde` and `/dev/sde1`, and continue for any other corruped disks.
+
+If your Rook-Ceph cluster does not have any critical data stored in it, it may be simpler to
+uninstall Rook completely and redeploy with v1.6.7 or higher.
+
+#### Handling randomly appearing partitions (v1.6.7+)
+If you are using Rook v1.6.7 or higher, it is safe to ignore these unexpected `atari` partitions.
+Rook will only deploy OSDs on `gpt` partitions.
```

Comment:
Is this true before v1.7?

Reply:
Good catch. I have to adjust the text for the 1.6 backport.